### PR TITLE
Adding openvdb to the superbuild

### PR DESCRIPTION
**Pull request by mwestphal**
- Adding openvdb project to the F3D superbuild, inspired from paraview-superbuild
 - Not enabled in macOS ARM for now, boost project is not working with it for now

- [x] CI artifacts Validated on Linux
- [x] CI artifacts Validated on Windows
- [x] CI artifacts validated on macOS

### CODE REVIEW

File: .github/actions/f3d-superbuild/action.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ runs:
       echo VTK_COMMIT_SHA=`cat ./vtk_commit_sha` >> $GITHUB_ENV
 
   # OCCT coloring is disabled on macOS because of f3d-app/f3d#792
+  # OpenVDB is disabled on macOS arm64 because boost iostreams does not supports it yet
```
Comment (Meakk) [inline .github/actions/f3d-superbuild/action.yml:77]: Is it an issue in common-superbuild or Boost itself?

File: projects/openvdb.cmake [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  set(openvdb_build_shared ${BUILD_SHARED_LIBS_openvdb})
+else ()
+  set(openvdb_build_shared ${BUILD_SHARED_LIBS})
+endif ()
+if (openvdb_build_shared)
+  set(openvdb_build_static OFF)
+  set(openvdb_msvc_runtime "MultiThreadedDLL")
+else ()
+  set(openvdb_build_static ON)
+  set(openvdb_msvc_runtime "MultiThreaded")
```
Comment (Meakk) [inline projects/openvdb.cmake:11]: why aren't we using `MultiThreadedDLL` for static lib too?

File: projects/vtk.cmake [hunk 3]
```diff
@@ -27,12 +32,23 @@ else ()
...
+  set(openvdb_build_shared ${BUILD_SHARED_LIBS_openvdb})
+else ()
+  set(openvdb_build_shared ${BUILD_SHARED_LIBS})
+endif ()
+if (openvdb_build_shared)
+  set(openvdb_build_static OFF)
+else ()
+  set(openvdb_build_static ON)
+endif ()
+
```
Comment (Meakk) [inline projects/vtk.cmake:45]: is it used here?

File: .github/actions/f3d-superbuild/action.yml (hunk 1 above)
Comment (mwestphal) [reply .github/actions/f3d-superbuild/action.yml:77]: the boost project in common-superbuild does not support it yet, i will clarify.

File: projects/patches/openvdb-remove-lib-prefix-static.patch [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+diff --git a/openvdb/openvdb/CMakeLists.txt b/openvdb/openvdb/CMakeLists.txt
+index 33d4e5c..2c16443 100644
+--- a/openvdb/openvdb/CMakeLists.txt
++++ b/openvdb/openvdb/CMakeLists.txt
+@@ -678,7 +678,6 @@ if(OPENVDB_CORE_STATIC)
+       target_compile_options(openvdb_static PUBLIC
+         "$<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/MT$<$<CONFIG:Debug>:d>>")
+     endif()
+-    set_target_properties(openvdb_static PROPERTIES PREFIX "lib")
```
Comment (Meakk) [inline projects/patches/openvdb-remove-lib-prefix-static.patch:18]: do we care about this patch?

Comment (mwestphal) [reply projects/patches/openvdb-remove-lib-prefix-static.patch:18]: we care a lot. Without it, the FindOpenVDB.cmake doesnt work on Windows with static openvdb.

Clearly untested code.

Comment (Meakk) [reply projects/patches/openvdb-remove-lib-prefix-static.patch:18]: ok ^^

**Review by Meakk — APPROVED** [review]

**Pull request merged**